File: SRXLite/Modules/AppSettings.vb
Option Strict On

Namespace Modules

    Public Module AppSettings

#Region " EncryptionKey "

        Public Function EncryptionKey() As String
            Return ConfigurationManager.AppSettings("EncryptionKey").ToString
        End Function

#End Region

#Region " Environment "

        Public Class Environment

            Public Shared Function IsDemo() As Boolean
                Return (configValue() = "DEMO")
            End Function

            Public Shared Function IsDev() As Boolean
                Return (configValue() = "DEV")
            End Function

            Public Shared Function IsProd() As Boolean
                Return (configValue() = "PROD")
            End Function

            Private Shared Function configValue() As String
                Return ConfigurationManager.AppSettings("Environment").ToUpper()
            End Function

        End Class

#End Region

#Region " FileSizeUploadLimit "

        Public Function FileSizeUploadLimit() As Integer
            Return IntCheck(ConfigurationManager.AppSettings("FileSizeUploadLimit"))
        End Function

        Public Function GetFileSizeUploadLimitMB() As Double
            Return GetFileSizeMB(FileSizeUploadLimit())
        End Function

        Public Function GetInitialFileSizeUploadLimitMB() As Double
            Return GetFileSizeMB(InitialFileSizeUploadLimit())
        End Function

        Public Function InitialFileSizeUploadLimit() As Integer
            Return IntCheck(ConfigurationManager.AppSettings("InitialFileSizeUploadLimit"))
        End Function

#End Region

#Region " IsEncryptionEnabled "

        Public Function IsEncryptionEnabled() As Boolean
            Return BoolCheck(ConfigurationManager.AppSettings("EncryptionEnabled"), True)
        End Function

#End Region

    End Module

End Namespace

File: SRXLite.CS/Modules/AppSettings.cs
using System;
using System.Configuration;
using static SRXLite.Modules.Util;

namespace SRXLite.Modules
{
    public static class AppSettings
    {
        #region EncryptionKey

        public static string EncryptionKey()
        {
            return ConfigurationManager.AppSettings["EncryptionKey"];
        }

        #endregion

        #region Environment

        public static class Environment
        {
            public static bool IsDemo()
            {
                return (ConfigValue() == "DEMO");
            }

            public static bool IsDev()
            {
                return (ConfigValue() == "DEV");
            }

            public static bool IsProd()
            {
                return (ConfigValue() == "PROD");
            }

            private static string ConfigValue()
            {
                return ConfigurationManager.AppSettings["Environment"].ToUpper();
            }
        }

        #endregion

        #region FileSizeUploadLimit

        public static int FileSizeUploadLimit()
        {
            return IntCheck(ConfigurationManager.AppSettings["FileSizeUploadLimit"]);
        }

        public static double GetFileSizeUploadLimitMB()
        {
            return GetFileSizeMB(FileSizeUploadLimit());
        }

        public static double GetInitialFileSizeUploadLimitMB()
        {
            return GetFileSizeMB(InitialFileSizeUploadLimit());
        }

        public static int InitialFileSizeUploadLimit()
        {
            return IntCheck(ConfigurationManager.AppSettings["InitialFileSizeUploadLimit"]);
        }

        #endregion

        #region IsEncryptionEnabled

        public static bool IsEncryptionEnabled()
        {
            return BoolCheck(ConfigurationManager.AppSettings["EncryptionEnabled"], true);
        }

        #endregion
    }
}
